src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php

```diff
@@ -91,6 +100,41 @@ public function finishView(FormView $view, FormInterface $form, array $options):
         $view->vars['attr'] = $attr;
     }
 
+    private function getUrlWithExtraOptions(string $url, array $extraOptions): string
+    {
+        $this->validateExtraOptions($extraOptions);
+
+        $extraOptions[self::CHECKSUM_KEY] = $this->checksumCalculator->calculateForArray($extraOptions);
+        $extraOptions = base64_encode(json_encode($extraOptions));
+
+        return sprintf(
+            '%s%s%s',
```

Comment:
Suprised to see sprintf "just" to append string... but I must admit it improves code readability

Reply:
I don't remember when I used the . for the last time to concatenate strings :D.